File: Back to Basics Approach 3rd Edition/cse_143/cs2_sections/section01_arrayList/markLength4.js
/**
 * @author Alex Golub
 * @since 21-Apr-21, 10:43 PM
 */
// const list = ["four", "for ", "all "];
const list = ['Don\'t', 'do', 'anything', 'to', 'the', 'list!'];
const temp = [];
for (let i = list.length - 1; i >= 0; i--) {
  temp.unshift(list[i]);
  if (list[i].length === 4) {
    temp.unshift('****');
  }
}
list.length = 0;
temp.forEach(word => list.push(word));
console.log(list);

File: Back to Basics Approach 3rd Edition/cse_143/cs2_sections/section01_arrayList/markLength4.java
package cse_143.cs2_sections.section01_arrayList;

import java.util.List;

/**
 * https://practiceit.cs.washington.edu/problem/view/cs2/sections/arraylist/markLength4
 *
 * @author Alex Golub
 * @since 21-Apr-21, 10:40 PM
 */
class markLength4 {
  public static void markLength4(List<String> list) {
    for (int i = list.size() - 1; i >= 0; i--) {
      if (list.get(i).length() == 4) {
        list.add(i, "****");
      }
    }
  }
}